### ISSUE TITLE
Issues Importing the Replay Into Blender(3.3.1)

### ISSUE THREAD

**Issue by Superkat32** [Sam54123/mc-world-export#14]
I just finished exporting a replay from Minecraft and tried importing the replay into Blender, but I got multiple issues. I'll try my best to describe everything and give as much helpful info as possible.

**The issue...**

Whenever I try to import a .replay file into Blender, it says this(I've shortened the file path text things)

```
Python: Traceback (most recent call last):
  File "Blender Foundation\Blender\3.3\scripts\addons\import_vcap\import_replay_operator.py", line 103, in execute
    replay_file.load_replay(self.filepath, context, context.scene.collection, handle=handle, settings=settings)

  File "Blender Foundation\Blender\3.3\scripts\addons\import_vcap\replay\replay_file.py", line 115, in load_replay
    with ZipFile(file, 'r') as archive:

  File "BlenderFoundation.Blender_3.3.1.0_x64__ppwjx1n5r4v9t\Blender\3.3\python\lib\zipfile.py", line 1258, in __init__
    self._RealGetContents()

  File "BlenderFoundation.Blender_3.3.1.0_x64__ppwjx1n5r4v9t\Blender\3.3\python\lib\zipfile.py", line 1325, in _RealGetContents
    raise BadZipFile("File is not a zip file")

zipfile.BadZipFile: File is not a zip file
```

**Hopefully some useful info...**

Minecraft version: 1.19.2
Fabric API version:  0.67.1
Replay Mod version: 2.6.8
Replay Exporter version: 0.4.0
Blender version: 3.3.1
<details><summary>Other mods I had installed(I couldn't get this to format nicely)...</summary>
<p>
3dSkinLayers-1.5.1, Appleskin-2.4.1, ArmorChroma-1.2.3, BetterF3-4.0.0, ClothConfig-8.2.88, EntityCulling-1.5.2, FerriteCore5.0.3, Iris-1.4.3, Lithium-0.10.2, Sodium-0.4.4, SodiumExtra-0.4.14, Starlight-1.1.1, WI-Zoom-1.4, ModMenu-4.1.1
</p>
</details>

I tried two more things related to other mods.
1. I tried recording with only four other mods, those being Fabric API, Mod Menu, Sodium, and Replay Mod. Still got the same issue when importing a new recording.
2. I then tried recording with the same mods, expect Sodium. This time, whenever I tried rendering the replay, it showed a different screen, with a black and white Replay Mod logo in the center of the screen, and a "frames rendered" bar at the bottom of the screen(I haven't seen this screen before). However, when trying to import it into Blender, it still had the same issue.
Each time I tried something new with other mods, I recorded a new replay.

Some info related to the replay itself(might be useful idk)...
The replay consists of me getting in and out of a boat quite a few times. I'm not sure if this would have an effect on the importing or not. I'm using the 45 degree rotated square icon and the sand timer like thing icon at the start and end of the replay(I believe those are the right buttons I should be pressing.) 

For the mod: I downloaded the mod from Modrinth.
For the addon: I downloaded the "import_vcap.zip" from the releases page, and then used the addon page in Blender to install it, like I have with other addons. The folder was still zipped.

Hopefully this is another info, and also hopefully I have this formatted decently enough for you to understand. If you need any more info, I'll try my best to get it to you.
Thanks in advance!

**Comment by Sam54123** [comment]
Can you send a copy of the file you're trying to import into Blender?

**Comment by Superkat32** [comment]
[theReplayFile.zip](https://github.com/Sam54123/mc-world-export/files/10147526/theReplayFile.zip)

Here you go! I've zipped up the .replay file by itself so that I can upload it to GitHub. Hopefully it works for ya. If you would like for me to upload the file in an alternative way, then please let me know(you would have to let me know where to upload it because I don't really know where I would upload it.)

**Comment by Sam54123** [comment]
So there's definitely something wrong with that file, but I don't think the problem comes from Minecraft. Looking at your file, it seems like the entire contents of the file have been wiped and replaced with the following ASCII:
```
BLENDER-v279
```
How exactly are you importing it into Blender? Because something is clearly going wrong on that front. The addon shouldn't be writing to the file at all, but there's no way Minecraft's exporting that string.

**Comment by Superkat32** [comment]
I'm going to describe everything I've done regarding the Blender addon, including the installation process.

After opening Blender, I pressed the edit button on the top of the screen, then Preferences. Within Preferences, I go to the addons page, then installed the "import_vcap.zip" from your releases page via the "Install..." button. I then enable the addon once it was done installing. Afterwards, I went to File, Import, "Minecraft Replay File (.replay)." Via the Blender's file explorer like window, then found the replay_videos folder within the .minecraft folder, and then pressed the file which I wanted to import(the same one I sent earlier) once, and then pressed the blue "Import Minecraft Replay" button.

Everything was checked except two options, the "Merge Vertices" and the "Separate Entity Parts" options are not checked.

After pressing the Import button, Blender gave me the error earlier described towards the button right of my screen.

I've tried closing and opening back up Blender, I've tried uninstalling and reinstalling the addon multiple times, and also tried installing the addon from the Beta 0.3.2 release, instead of the 0.4.0 release, though it gave the same error.

Something I did notice which might be helpful, is that the three replays I've recorded are all 12B according to Blender, despite the lengths being off by a few seconds. Roughly the same stuff happens in each replay, though each one is slightly different. I'm not sure if they were that size before attempting to import it into Blender, but I could try recording another replay, then checking that if you would like me too.

**Comment by Sam54123** [comment]
Yeah, try exporting another replay. The file you sent me is corrupted, so it won't import even if you do anything right. Also, back it up before you import it. That way, if Blender *is* re-writing the file somehow, we still have the original.

**Comment by Superkat32** [comment]
I'm rerecording the replay again(coming soon), but while doing so I remembered I don't _think_ I have FFmpeg setup, do you think that could be it?

**Comment by Superkat32** [comment]
[moreReplayFiles.zip](https://github.com/Sam54123/mc-world-export/files/10149312/moreReplayFiles.zip)

Here are two more replay files. "test3" has been imported into Blender. "test3backup" has not been imported into Blender.

The more I think about it though, I really think it could be the fact I don't have FFmpeg setup that could be causing this. I'll try to set it up soon and let you know what happens, but I won't be able to set it up today.

**Comment by Sam54123** [comment]
> I'm rerecording the replay again(coming soon), but while doing so I remembered I don't have FFmpeg setup(Replay Mod said I couldn't export the camera paths normally, so that means I 100% don't have it setup), do you think that could be it?


No. The export system has nothing to do with ffmpeg

**Comment by Sam54123** [comment]
Actually.... it might

**Comment by Sam54123** [comment]
Now that I think about how the replay export system works, and the hacks I used to get the Replay Mod to use it, that could be the case.

**Comment by Superkat32** [comment]
Alright. I'll let you know whenever I get FFmpeg installed, and if it fixes the issue or not. I'm very sorry for any frustration this may have caused because of a mistake on my part.

**Comment by Sam54123** [comment]
Oh, no worries! Thanks for bringing this to my attention!

**Comment by Sam54123** [comment]
Did it work?

**Comment by Superkat32** [comment]
Sorry for the delay, but I have gotten FFmpeg installed and setup for Replay Mod, and the exporting works beautifully! I only  had a couple issues beyond the original issue, one of which was on the original test world. Newly exported .replay file still was only 12B, so I decided to create a new test world just in case and it worked! (I guess the only thing you have to do is add a "work pls" in the world name... /s). The other issue I had was trying to figure out how to change the skin within Blender. I thought it would be similar to changing a skin with MCPrep, but I was able to figure it out with enough time and tinkering.

Thanks again for both helping me fix the issue and creating the super cool mod! I have a feeling I'm going to have a LOT of fun with this thing.

**Comment by Sam54123** [comment]
@Superkat32 Can you verify if this still happens in 0.7.0?

**Comment by Superkat32** [comment]
Yes! I will try to see if it still happens. I'll need some time though, so it'll probably happen either very late tonight, tomorrow, or this weekend.

**Comment by Superkat32** [comment]
Hello! I'm really sorry for being late, I completely forgot. However, I tested it, and it works perfectly fine without FFmpeg! Replay Mod made the little sound effect, and I was able to import it into Blender without seemingly any issues popping up!
I would like to note that I was using a different device for testing this, a Windows 11. I'm not sure if Windows 11 comes with FFmpeg installed automatically, but after a quick Google search, I don't think it does.
